samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java

```diff
@@ -1645,6 +1695,42 @@ public String getBlockDivs() {
 	  return selections;
   }
 
+    public void setCategoriesEnabled(boolean categoriesEnabled) {
+        this.categoriesEnabled = categoriesEnabled;
+    }
+
+    public boolean getCategoriesEnabled() {
+        return categoriesEnabled;
+    }
+
+    /**
+     * Populate the categoriesSelectList property with a list of string names
+     * of the categories in the gradebook
+     */
+    private List populateCategoriesSelectList() {
+        List<CategoryDefinition> categoryDefinitions;
+        List<SelectItem> selectList = new ArrayList<>();
+
+        if (this.gradebookExists) {
+            GradebookService g = (GradebookService)  ComponentManager.get("org.sakaiproject.service.gradebook.GradebookService");
```

Comment:
should the gradebook service be injected as a member variable vs being looked up, I am not sure the frequency on this but if it's whenever the gradebook tool exists in the site then it's probably pretty often.

I am not a fan of how Samigo does spring dependency lookups via the IntegrationContextFactory.

Using the ComponentManager cover also has its drawbacks (though its simplicity is nice).

Since this is in the webapp you could use the pattern from ExportResponsesBean (see getAutowireCapableBeanFactory) to use spring to autowire it which is probably better than both of the above as it more aligns with how it should be done.

Reply:
Thanks, Earle. I liked the idea of using Spring to autowire the dependencies and ExportResponsesBean was a good model. I've changed both the assessment settings and published settings beans to follow this pattern. I took the opportunity to also change tool manager and session manager to do the same thing instead of use the cover.

I left everything in a separate commit so that it would be easier to review because it's not something I've done before. Testing seemed fine though. Let me know if you have any feedback.